File: csharp/Database/DbConfig.cs
using Cassandra;
using Newtonsoft.Json.Linq;
using System;
using System.Linq;
using System.Threading.Tasks;

namespace csharp.Database {
   public class DbConfig {
       ISession session;
       
       public DbConfig() {
           // Connect to the demo keyspace on our cluster running at 127.0.0.1
           Cluster cluster = Cluster.Builder().AddContactPoint("127.0.0.1").Build();
           session = cluster.Connect("benchmark");
       }
    
       public void insertTransactionAsync(dynamic data) {
           Task.Run(() => {
               JObject json = JObject.FromObject(new {
                    // generate UUID
                    code = Guid.NewGuid(),
                    tags = data.tags,
                    values = data.values,
                    address = data.address
               });

               string sql = "insert into transactions JSON '"+json.ToString()+"'";
               //Console.Write(sql);

               session.Execute(sql);
           });
       }
        
       public Task<int> findTagAsync(string key) {
           return Task.Run<int>(() => {
               string sql = "select value from tags where key = '"+key+"'";
               Row result = session.Execute(sql).First();

               string svalue = result["value"].ToString();
               int value1 = 0;
               if (int.TryParse(svalue, out value1)) {
                   return value1;
               } else {
                   return -1;
               }
           });
       }
   }
}

File: node/src/db-config.js
'use strict';
var cassandra = require('cassandra-driver');
var uuid = require('node-uuid');
var Q = require('q');

var client = new cassandra.Client({contactPoints: ['127.0.0.1'], keyspace: 'benchmark'});

exports.insertTransaction = (data, cb) => {
    var json = {
        // generate UUID
        code: uuid.v1(),
        tags: data.tags,
        values: data.values,
        address: data.address
    };
    
    let deferred = Q.defer(),
        query = [{
            query: 'insert into transactions JSON ?',
            params: [JSON.stringify(json)]
        }];
    client.batch(query, {prepare: true}, (err) => {
        if (err) {
            deferred.reject(new Error(err));
            console.error(err);
        } else {
            deferred.resolve();
        }
    });
    return deferred.promise;
};

exports.findTag = function (key) {
    let deferred = Q.defer(),
        query = 'select value from tags where key = ?';
    client.execute(query, [key], (err, result) => {
        if (err) {
            deferred.reject(new Error(err));
            console.error(err);
        } else {
            deferred.resolve(result.rows[0].value);
        }
    });
    return deferred.promise;
};